helix-core/src/main/java/org/apache/helix/ConfigAccessor.java

```diff
@@ -123,9 +123,22 @@ public ConfigAccessor(HelixZkClient zkClient) {
    * @param zkAddress
    */
   public ConfigAccessor(String zkAddress) {
-    _zkClient = SharedZkClientFactory.getInstance()
-        .buildZkClient(new HelixZkClient.ZkConnectionConfig(zkAddress),
-            new HelixZkClient.ZkClientConfig().setZkSerializer(new ZNRecordSerializer()));
+    // First, attempt to connect on multi-realm mode using FederatedZkClient
+    RealmAwareZkClient zkClient;
+    try {
+      zkClient = new FederatedZkClient(
+          new RealmAwareZkClient.RealmAwareZkConnectionConfig.Builder().build(),
+          new RealmAwareZkClient.RealmAwareZkClientConfig());
+    } catch (IOException | InvalidRoutingDataException e) {
```

Comment:
`IllegalStateException` floated from HttpRoutingDataReader should also be caught.

Reply:
Good catch! Got those sharp eyes on you :)